saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/metadata/SAMLRoleDescriptorParser.java

```diff
@@ -0,0 +1,37 @@
+package org.keycloak.saml.processing.core.parsers.saml.metadata;
+
+import org.keycloak.dom.saml.v2.metadata.RoleDescriptorType;
+import org.keycloak.saml.common.exceptions.ParsingException;
+import org.keycloak.saml.common.util.StaxParserUtil;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.events.StartElement;
+
+import java.util.List;
+
+import static org.keycloak.saml.processing.core.parsers.saml.metadata.SAMLMetadataQNames.ROLE_DESCRIPTOR;
+
+/**
+ * @author mhajas
+ */
+public class SAMLRoleDescriptorParser extends SAMLRoleDecriptorTypeParser<RoleDescriptorType> {
```

Comment:
Should not be here - "The <RoleDescriptor> element is an **abstract** extension point" (saml-metadata-2.0.pdf, line 551)